testing/async/inclusive_scan/mixed_types.cu

```diff
@@ -0,0 +1,95 @@
+#include <thrust/detail/config.h>
+
+#if THRUST_CPP_DIALECT >= 2014
+
+#include <async/test_policy_overloads.h>
+
+#include <async/inclusive_scan/mixin.h>
+
+// Test using mixed int/float types for:
+// - input_value_type       | (int, float)
+// - output_value_type      | (int, float)
+// - thrust::plus<T> T-type | (int, float, void>)
+//
+// The thrust::plus<T> types are covered by the
+// scan_mixed_types_overloads component.
+//
+// The testing/scan.cu TestMixedTypes test spells out the expected behavior,
+// which is defined by https://wg21.link/P0571.
+
+namespace
+{
+
+template <typename value_type>
+struct mixed_type_input_generator
+{
+  using input_type = thrust::device_vector<value_type>;
+
+  static input_type generate_input(std::size_t num_values)
+  {
+    input_type input(num_values);
+    thrust::sequence(input.begin(),
+                     input.end(),
+                     // fractional values are chosen deliberately to test
+                     // casting orders and accumulator types:
+                     static_cast<value_type>(1.5),
+                     static_cast<value_type>(1));
+    return input;
+  }
+};
+
+// A fractional value is used to ensure that a different result is obtained when
+// using float vs. int.
+struct mixed_types_postfix_args
+{
+  using postfix_args_type = std::tuple<  // Overloads to test:
+    std::tuple<>,                        // - no extra args
+    std::tuple<thrust::plus<>>,          // - plus<>
+    std::tuple<thrust::plus<int>>,       // - plus<int>
+    std::tuple<thrust::plus<float>>      // - plus<float>
+    >;
+
+  static postfix_args_type generate_postfix_args()
+  {
+    return {{},
+            {thrust::plus<>{}},
+            {thrust::plus<int>{}},
+            {thrust::plus<float>{}}};
+  }
+};
+
+template <typename input_value_type,
+          typename output_value_type>
+struct invoker
+    : mixed_type_input_generator<input_value_type>
+    , testing::async::mixin::output::device_vector<output_value_type>
+    , mixed_types_postfix_args
+    , testing::async::inclusive_scan::mixin::invoke_reference::
+        host_synchronous<input_value_type, output_value_type>
+    , testing::async::inclusive_scan::mixin::invoke_async::basic
+    , testing::async::mixin::compare_outputs::assert_maybe_fuzzy_equal
+{
+  static std::string description()
+  {
+    return "mixed input/output/functor value_type tests";
+  }
+};
+
+} // namespace
+
+void TestScanMixedTypes()
+{
+  // 10 values are enough to check the behavior we want to test while staying
+  // small enough to reason about during debugging.
+  constexpr std::size_t num_values = 10;
```

Comment:
This is incorrect, because CUB has different codepaths for single tile versus multi-tile; we need to test both, so this should use the testing mechanism that supports a variable problem size (ditto for the corresponding exclusive_scan test).

Reply:
I added a `DECLARE_SIZED_UNITTEST` macro and used it for these. The new macro was needed since the others all expect the test to take template parameters.